### PR TITLE
Refactor gerrit client so that auth logic can be reused

### PR DESCRIPTION
**Pull request by chaodaiG**

### CODE REVIEW

**Comment by chaodaiG** [comment]
flaky test, deflaking in #27152 
/retest

File: prow/gerrit/client/client.go [hunk 1]
```diff
@@ -204,6 +204,36 @@ func NewClient(instances map[string]map[string]*config.GerritQueryFilter) (*Clie
...
+	go func() {
+		for {
+			c.applyGlobalConfigOnce(cfg, lastSyncTracker, cookiefilePath, tokenPathOverride, additionalFunc)
+			// No need to spin constantly, give it a break. It's ok that config change has one second delay.
+			time.Sleep(time.Second)
+		}
+	}()
+}
+
+func (c *Client) applyGlobalConfigOnce(cfg config.Getter, lastSyncTracker *SyncTime, cookiefilePath, tokenPathOverride string, additionalFunc func()) {
```
Comment (listx) [inline prow/gerrit/client/client.go:219]: Looks like `additionalFunc` param is not used in the function body. Remove?

**Review by listx — COMMENTED** [review]
I guess this dedupes logic? Is that the main goal here or is there some other reason that you had in mind?

File: prow/gerrit/client/syncer.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ limitations under the License.
 */
 
 // Package adapter implements a controller that interacts with gerrit instances
```
[the remark this answers is not in the document]
Comment (chaodaiG) [reply prow/gerrit/client/syncer.go]: done

File: prow/gerrit/client/client.go (hunk 1 above)
Comment (chaodaiG) [reply prow/gerrit/client/client.go:219]: it's actually used, added

**Comment by chaodaiG** [comment]
> I guess this dedupes logic? Is that the main goal here or is there some other reason that you had in mind?

yeah mainly dedupes purpose. This same code path is being repeated here and will be used in tide as in #27146 , this PR should help duplicating very similar code multiple places

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**